staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -671,6 +673,24 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+// Returns true if the given err is transient networking error that should be retried.
+// Currently 'connection refused' and 'connection reset' errors will be retried.
+func IsRetriableConnectionError(err error) bool {
```

Comment:
this lacks context ("was this a GET request") to know if this error is able to be retried, and is confusing since it doesn't check retry-after headers.

I still think this would be better as a new "net.IsConnectionRefused()" function, used inline below as `if net.IsConnectionReset(err) || net.IsConnectionRefused(err) { ... }`, and not exported here

Reply:
still outstanding

Reply:
@liggitt `net.IsConnectionRefused` does not exists in go `net` package, would you be ok with private function here?

Reply:
the net package we're using here is our util package:

https://github.com/kubernetes/kubernetes/blob/1d441c1f93e7cc44f8a200df28a2c8a4bee3a2bb/staging/src/k8s.io/apimachinery/pkg/util/net/util.go#L41-L56

Reply:
🤦‍♂️ sure thing.